clients/include/testing_logging.hpp

```diff
@@ -0,0 +1,306 @@
+/* ************************************************************************
+ * Copyright 2016 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include <stdlib.h>
+#include <vector>
+
+#include "rocblas.hpp"
+#include "utility.h"
+#include "cblas_interface.h"
+#include "norm.h"
+#include "unit.h"
+#include "arg_check.h"
+#include <complex.h>
+#include <unistd.h>
+#include <pwd.h>
+#include <fstream>
+#include <string>
+#include <algorithm>
+#include <iterator>
+
+using namespace std;
+
+// replaces X in string with s, d, c, z or h depending on typename T
+template <typename T>
+std::string replaceX(std::string input_string)
+{
+    if(std::is_same<T, float>::value)
+    {
+        std::replace(input_string.begin(), input_string.end(), 'X', 's');
+    }
+    else if(std::is_same<T, double>::value)
+    {
+        std::replace(input_string.begin(), input_string.end(), 'X', 'd');
+    }
+    else if(std::is_same<T, rocblas_float_complex>::value)
+    {
+        std::replace(input_string.begin(), input_string.end(), 'X', 'c');
+    }
+    else if(std::is_same<T, rocblas_double_complex>::value)
+    {
+        std::replace(input_string.begin(), input_string.end(), 'X', 'z');
+    }
+    else if(std::is_same<T, rocblas_half>::value)
+    {
+        std::replace(input_string.begin(), input_string.end(), 'X', 'h');
+    }
+    return input_string;
```

Comment:
Maybe you could document what characters we should use for complex half types.  Could we instead use of the proposed codes, such as r16?

Reply:
I think  h is commonly used for half real, so we can keep with it. I have not seen demand for half complex. Can we delay deciding what we do for half complex till we need it, rather than specifying now what we will use.